filters/auth/tokenintrospection.go

```diff
@@ -185,6 +241,12 @@ func (s *tokenIntrospectionSpec) CreateFilter(args []interface{}) (filters.Filte
 		issuerAuthClient[issuerURL] = ac
 	}
 
+	if s.secure && clientId != "" && clientSecret != "" {
+		ac.url.User = url.UserPassword(clientId, clientSecret)
+	} else {
+		ac.url.User = nil
```

Comment:
I think the default is nil already, so else part could be dropped.

Reply:
I added this to fix a caching issue. 

In testing, I found that if I created a filter with credentials:
eg. secureOauthTokenintrospectionAllKV("https://provider.url", "client-id", "client-secret", "key", "value")

and then subsequently updated the filter to use one without credentials:
oauthTokenintrospectionAllKV("https://provider.url", "key", "value")

if would cache the config and allow the request to go through even though the introspection endpoint requires authentication.

It was one of those use-cases that might not be very common but could lead to some unexplained behavior.